### DISCUSSION TITLE
aws lambdas, internal packages and the npm serverless utility

### DISCUSSION THREAD

**Discussion by pedrohba1** [vercel/turborepo#5905] [Help] [answered]
I'm having a hard time making lambdas get the packages of my monorepo as dependencies and I would like some instruction on what I should do in this scenario below. Basically, the code that is in my packages do not reach the `node_modules` of the `main-service`, becase the`@monorepo/my-package` is not inserted inside the dependencies . I'm developing with nestjs and typescript, if this information matters.

I'm trying to make a serverless apllication, and I need to share some code between multiple lambdas. 
This code for example:

```ts
import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
import { Document } from 'mongoose';
import * as mongoose from 'mongoose';

export type WalletDocument = Wallet & Document;

@Schema()
export class Wallet {
  _id: mongoose.Schema.Types.ObjectId;
  @Prop({ unique: true, required: true, sparse: true })
  address: string;
  @Prop({ unique: true, required: true, sparse: true })
  hexadecimal: string;
  @Prop({ required: true, default: '0' })
  balance: string;
}

export const WalletSchema = SchemaFactory.createForClass(Wallet);
```

I added it to a `packages/shared-models` and I'm exporting these in a way that I can use it in `main-service` like this:

```ts
import { Wallet, WalletSchema } from '@monorepo/shared-models';
```

And this code is added in the `package.json` of `main-service` as a depdency too:

```json
  "dependencies": {
    "@monorepo/shared-models": "1.0.0",
   ...
```
It works fine for running the application locally.

The problem is, when I deploy this lambda, I have this error being returned:

```
"errorType": "Runtime.ImportModuleError",
    "errorMessage": "Error: Cannot find module '@monorepo/shared-models'\nRequire stack:\n- /var/task/src/users/users.module.js\n- /var/task/src/app.module.js\n- /var/task/src/lambda.js\n- /var/runtime/index.mjs",
    "stack": [
```

Which means that it cannot find the `@monorepo/shared-models` as a dependency. And in fact, it is not located in the `node_modules` of `main-service`.

I'm looking for a way to solve this problem. What can I do for my serverless function to receive the code in `@monorepo/shared-models`?

**Comment by anthonyshew** [comment, marked as answer]
Closing as a dupe of #7009
